pkg/agent/flexvolume/controller.go

```diff
@@ -196,23 +194,35 @@ func (c *FlexvolumeController) Detach(detachOpts AttachOptions, _ *struct{} /* v
 	return nil
 }
 
-// RemoveAttachmentObject removes the attachment from the VolumeAttachment CRD
-func (c *FlexvolumeController) RemoveAttachmentObject(detachOpts AttachOptions, _ *struct{} /* void reply */) error {
+// RemoveAttachmentObject removes the attachment from the VolumeAttachment CRD and returns whether the volume is safe to detach
+func (c *FlexvolumeController) RemoveAttachmentObject(detachOpts AttachOptions, safeToDetach *bool) error {
 	namespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
 	crdName := detachOpts.VolumeName
 	logger.Infof("Deleting attachment for mountDir %s from Volume attach CRD %s/%s", detachOpts.MountDir, namespace, crdName)
 	volumeAttach, err := c.volumeAttachmentController.Get(namespace, crdName)
 	if err != nil {
 		return fmt.Errorf("failed to get Volume attach CRD %s/%s: %+v", namespace, crdName, err)
 	}
+	node := os.Getenv(k8sutil.NodeNameEnvVar)
+	nodeAttachmentCount := 0
+	needUpdate := false
 	for i, v := range volumeAttach.Attachments {
-		if v.MountDir == detachOpts.MountDir {
-			// Deleting slice
-			volumeAttach.Attachments = append(volumeAttach.Attachments[:i], volumeAttach.Attachments[i+1:]...)
+		if v.Node == node {
+			nodeAttachmentCount++
+			if v.MountDir == detachOpts.MountDir {
+				// Deleting slice
+				volumeAttach.Attachments = append(volumeAttach.Attachments[:i], volumeAttach.Attachments[i+1:]...)
+				needUpdate = true
```

Comment:
instead of using an extra boolean variable, could you only increment `nodeAttachmentCount` if the mount dirs do not match?  i think that's basically the intent of what you are doing (not counting the attachment we are getting rid of).  that may be easier to read instead of bringing in this extra bool var.

you'd also change the below check to be `if nodeAttachmentCount == 0`

Reply:
I also use this variable to get the number of attachments left on this node and based on this, it will set the safeTodetach  var

Reply:
right, i believe that's what i was referring to with the note above about if you took my suggestion, you'd need to make a change like so too: `if nodeAttachmentCount == 0`.

but it doesn't matter, the way you have it is not wrong, i just thought it'd be easier to read.

Reply:
I was incorrect, we are indeed tracking two different things here.  Just the lone `nodeAttachmentCount` var by itself would not be enough.